src/pix/client.rs

```diff
@@ -99,22 +146,49 @@ impl Client {
         }
     }
 
+    /// Flush the write buffer.
+    pub fn flush(&mut self) -> Result<(), Error> {
+        self.stream.flush()?;
+        self.current_size = 0;
+        Ok(())
+    }
+
     /// Write the given command to the given stream.
     fn write_command(&mut self, cmd: &[u8], newline: bool) -> Result<(), Error> {
         // Write the pixels and a new line
+
+        let new_size: u16 = match self.flush_mode {
+            FlushMode::Manual => {
+                self.stream.write_all(cmd)?;
+                if newline {
+                    self.stream.write_all(b"\n")?;
+                }
+                return Ok(());
+            }
+            FlushMode::Bytes => {
+                let mut new_size = cmd.len();
+                if newline {
+                    new_size += 1;
+                }
+                new_size as u16
+            }
+            FlushMode::Commands => 1,
+        };
+
+        if self.current_size + new_size > self.flush_size {
```

Comment:
Shouldn't we flush if equal as well?

```suggestion
        if self.current_size + new_size >= self.flush_size {
```